doc/sphinx/language/gallina-extensions.rst

```diff
@@ -951,83 +944,39 @@ Reserved commands inside an interactive module
     .. exn:: This is not the last opened module.
        :undocumented:
 
-.. cmd:: Module @ident := @module_expression
-
-    This command defines the module identifier :token:`ident` to be equal
-    to :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` and body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} : @module_type := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` (possibly none), and output module type :token:`module_type`,
-       with body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
-
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
-
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
-
-       is equivalent to an interactive module where each :token:`module_expression` is included.
-
-.. cmd:: Module Type @ident
-
-   This command is used to start an interactive module type :token:`ident`.
-
-   .. cmdv:: Module Type @ident {* @module_binding}
-
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
-
 
 Reserved commands inside an interactive module type:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd:: Include @module
-
-   Same as ``Include`` inside a module.
-
-.. cmd:: Include {+<+ @module}
-
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
-
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
-
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
-
-.. cmd:: End @ident
+.. cmdv:: End @ident
 
    This command closes the interactive module type :token:`ident`.
 
    .. exn:: This is not the last opened module type.
       :undocumented:
 
-.. cmd:: Module Type @ident := @module_type
+.. cmd:: Module Type @ident {* @module_binder } {+ <: @module_type_inl } {? @is_module_type }
 
-   Defines a module type :token:`ident` equal to :token:`module_type`.
+   .. insertprodn is_module_type is_module_type
 
-    .. cmdv:: Module Type @ident {* @module_binding} := @module_type
+   .. prodn::
+      is_module_type ::= := @module_type_inl {* <+ @module_type_inl }
```

Comment:
I suggest inlining this or renaming to `module_type_body` (but I think I prefer the former).